src/main/java/strategy/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package strategy;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy operationStrategy;
+    private Storage storage;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+        this.storage = new Storage();
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            operationStrategy.getHandler(transaction.getOperation());
```

Comment:
You are retrieving the handler from the `operationStrategy` but not executing any operation with it. You need to call a method on the handler to actually perform the transaction operation.